### ISSUE TITLE
Discord.NET 2.0 breaks optional arguments in commands before the remainder?

### ISSUE THREAD

**Issue by Aida-Enna** [RogueException/Discord.Net#1221]
```
[Command("backupemotes", RunMode = RunMode.Async)]
[RequireOwner]
public async Task BackupEmotesAsync(bool AttachResults = true, [Remainder][Summary("Server ID")] string ServerIDString = "")
{
}
```

This worked fine in (whatever was before 2.0) through ```!backupemotes``` or ```!backupemotes 81384788765712384```, the latter being a serverID
Now when I do the first (!backupemotes) it works, but the second (!backupemotes 81384788765712384) says it's failing to parse the boolean - but it should be skipping the boolean and only giving the remainder to the ServerIDString.

**Comment by Aida-Enna** [comment]
Actually it looks like it may be the reverse - This may be a <2.0 bug that was fixed in 2.0, which is why it no longer works.

**Comment by FiniteReality** [comment]
Yeah. It seems like the way it is parsing now is the correct, intentional way.
